packages/lesswrong/testing/moderation/moderation.frontend.tests.js

```diff
@@ -0,0 +1,204 @@
+      />)
+    expect(commentsItem.find(".comments-item-reply-link")).to.have.length(0);
+  });
+  it('CommentsItem renders reply-button when user is in a Post bannedUserIds list but PostAuthor NOT in trustLevel1', async () => {
+    const user = await createDummyUser()
+    const author = await createDummyUser()
+    const post = await createDummyPost(author, {bannedUserIds:[user._id]})
+
+    const commentsItem = shallow(
+      <CommentsItem
+        {...commentMockProps.commentMockProps}
+        currentUser={user}
+        post={post}
+      />)
+    expect(commentsItem.find(".comments-item-reply-link")).to.have.length(1);
+  });
+  it('CommentsItem does NOT render reply-button when user is in a User bannedUserIds list', async () => {
+    const user = await createDummyUser()
+    const author = await createDummyUser({groups:['trustLevel1'], bannedUserIds:[user._id]})
+    const post = await createDummyPost(author)
+    const commentsItem = shallow(
+      <CommentsItem
+        {...commentMockProps.commentMockProps}
+        currentUser={user}
+        post={post}
+      />)
+    expect(commentsItem.find(".comments-item-reply-link")).to.have.length(0);
+  });
+  it('CommentsItem renders reply-button when user is in Post bannedUserIds list but User is NOT in trustLevel1', async () => {
+    const user = await createDummyUser()
+    const author = await createDummyUser({bannedUserIds:[user._id]})
+    const post = await createDummyPost(author)
+
+    const commentsItem = shallow(
+      <CommentsItem
+        {...commentMockProps.commentMockProps}
+        currentUser={user}
+        post={post}
+      />)
+    expect(commentsItem.find(".comments-item-reply-link")).to.have.length(1);
+  });
+  it('commentsListSection renders new-comment-form when user is NOT in a Post bannedUserIds list', () => {
+    const commentsListSection = shallow(
+      <CommentsListSection
+        { ...commentListMockProps}
+        currentUser={{_id:"1234"}}
+        post={{id:"",slug:"", bannedUserIds:[], user:{groups:['trustLevel1']}}}
+      />,
+      {context:{client:mockClient}}
+    )
+    expect(commentsListSection.find(".posts-comments-thread-new")).to.have.length(1);
+    const commentsListSection2 = shallow(
+      <CommentsListSection
+        { ...commentListMockProps}
+        currentUser={{_id:"12345"}}
+        post={{id:"",slug:""}}
+      />,
+      {context:{client:mockClient}}
+    )
+    expect(commentsListSection2.find(".posts-comments-thread-new")).to.have.length(1);
+  });
+  it('commentsListSection does NOT render new-comment-form when user is in a Post bannedUserIds list', () => {
+    const commentsListSection = shallow(
+      <CommentsListSection
+        { ...commentListMockProps}
+        currentUser={{_id:"12345"}}
+        post={{bannedUserIds:["12345"], user:{groups:['trustLevel1']}}}
+      />,
+      {context:{client:mockClient}}
+    )
+    expect(commentsListSection.find(".posts-comments-thread-new")).to.have.length(0);
```

Comment:
Is there a reason you are using the `expect` paradigm in all of these tests, as opposed to the should paradigm?

Reply:
I found the error messages clearer - sometimes a function would return undefined, and then the error would be "should is not a method of undefined", whereas with expect, the error is "undefined !== true"